xgcm/grid.py

```diff
@@ -733,6 +742,17 @@ def _wrap_and_replace_coords(self, da, data_new, position_to):
                 if d in da.coords:
                     coords[d] = da.coords[d]
 
+        # add compatible coords
+        if keep_coords:
+            # v0: compact but not readable version:
+            # coords.update(**{c: da[c] for c in da.coords
+            #                 if c not in coords and all([d in coords for d in da[c].dims])})
+            # v1: less compact but more readable version:
+            for c in da.coords:
+                if c not in coords and all([d in coords for d in da[c].dims]):
```

Comment:
This could be reused for the test.

Reply:
Also I think you are right, and this version is more readable. So I recommend going with this.